### PR TITLE
Adds Entrypoint data type

### PR DESCRIPTION
**Pull request by denibertovic**
@jprider63 @kseo how does this look to you? 

(please ignore the formatting changes in tests.hs and just focus on `testEntrypointJson`)

### CODE REVIEW

**Comment by denibertovic** [comment]
this is work on top of #22

File: src/Docker/Client/Types.hs [hunk 1]
```diff
@@ -643,7 +645,7 @@ defaultContainerConfig imageName = ContainerConfig {
                      , cmd=[]
                      , volumes=[]
                      , workingDir=Nothing
-                     , entrypoint=Nothing
+                     , entrypoint=NoEntrypoint
```
Comment (denibertovic) [inline src/Docker/Client/Types.hs]: The only thing that comes to mind is will this correctly override an entrypoint of and existing image. Ie. if we wan't to change the entrypoint of and image to null, effectively stripping the image of it's entrypoint (for some reason) ... I'm not actually sure this is possible with docker at all. Also is this a sane default?

**Comment by jprider63** [comment]
One thing I might change is drop the `NoEntrypoint` constructor. This looks equivalent to `Entrypoint []` to me. Is there any reason to include it? 

Otherwise, looks good to me!

**Comment by denibertovic** [comment]
@jprider63 good point. I removed the NoEntrypoint constructor. Merging.

**Pull request merged**